uro98/assignment2/BinaryTree.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * Represents a generic binary tree.
+ *
+ */
+public class BinaryTree<T> {
+  
+  Node<T> root;
+  
+  /**
+   * Set the root node to the value specified.
+   * @param data a generic value
+   */
+  public void addRoot(T data) {
+    Node<T> newNode = new Node<T>(data, 0);
+    
+    if (root == null) {
```

Comment:
Ideally this check should be done before creating node and if `root` is not `null`, something like `IllegalStateException` was thrown to signal error.